src/App.tsx

```diff
@@ -1,61 +1,100 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
 
 export const App = () => {
+  const [text, setText] = useState('');
+  const [user, setUser] = useState(0);
+  const users = [...usersFromServer];
+  const [todos, setTodos] = useState([...todosFromServer]);
+  const [errorText, setErrorText] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+  const [touched, setTouched] = useState(false);
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+    setErrorText((touched && !text) || (text.trim() === ''));
+    setErrorUser(user === 0);
+
+    if (!touched || (touched && (!text || text.trim() === ''))
+      || (touched && user === 0)) {
+      return;
+    }
+
+    const newIdComment = Math.max(...todos.map((todo) => todo.id), 0) + 1;
+    const newComment = {
+      id: newIdComment,
+      title: text,
+      completed: false,
+      userId: user,
+    };
+
+    setTodos([...todos, newComment]);
+    setText('');
+    setUser(0);
+    setErrorText(false);
+    setErrorUser(false);
+    setTouched(false);
+  }
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="text">Title:</label>
+          <input
+            id="text"
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            value={text}
+            onChange={event => {
+              setText(event.target.value);
+              setErrorText(false);
+            }}
+            onBlur={() => setTouched(true)}
+          />
+          {errorText && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
+          <label htmlFor="user">User:</label>
+          <select
+            id="user"
+            data-cy="userSelect"
+            value={user}
+            onChange={event => setUser(+event.target.value)}
+            onBlur={() => {
+              setTouched(true);
+              setErrorUser(false);
+            }}
```

Comment:
Move the onBlur logic to a separate function. It follows the best practice of separating concerns and keeping your JSX code clean and focused on rendering components rather than defining complex logic directly within it.